Sources/Adapters/Tensorflow/TFConverter+Mappers.swift

```diff
@@ -63,25 +63,51 @@ public extension TFConverter {
-                                    strideY: Int(strides.y))
-            return Convolution(convSize: convSize,
-                               neuronType: node.nodeDef.activationNeuron(),
-                               useBias: weightData.useBias,
-                               padding: PaddingType.fromTF(padString),
-                               weights: weights,
-                               bias: weightData.bias,
-                               id: node.nodeDef.name)
+
+            if node.nodeDef.isTFDepthwiseConvOp, #available(iOS 11.0, *) {
+                //transpose weights
+                let weights = weightData.weights != nil ? HWIOtoIOWH(weights: weightData.weights!, shape: weightData.weightShape.toShape) : nil
+                // Depthwise does not change feature channels count
+                let convSize = ConvSize(outputChannels: weightData.weightShape.inputChannels,
+                                        kernelWidth: weightData.weightShape.kernelHeight,
+                                        kernelHeight: weightData.weightShape.kernelWidth,
+                                        strideX: Int(strides.x),
+                                        strideY: Int(strides.y),
+                                        dilationX: dilations.x,
+                                        dilationY: dilations.y)
+                return DepthwiseConvolution(convSize: convSize,
+                                            neuronType: node.nodeDef.activationNeuron(),
+                                            useBias: weightData.useBias,
+                                            padding: PaddingType.fromTF(padString),
+                                            weights: weights,
+                                            bias: weightData.bias,
+                                            id: node.nodeDef.name)
+            } else {
+                //transpose weights
+                let weights = weightData.weights != nil ? HWIOtoOHWI(weights: weightData.weights!, shape: weightData.weightShape.toShape) : nil
+                let convSize = ConvSize(shape: weightData.weightShape,
+                                        strideX: Int(strides.x),
+                                        strideY: Int(strides.y),
+                                        dilationX: dilations.x,
+                                        dilationY: dilations.y)
+                return Convolution(convSize: convSize,
+                                   neuronType: node.nodeDef.activationNeuron(),
+                                   useBias: weightData.useBias,
+                                   padding: PaddingType.fromTF(padString),
+                                   weights: weights,
+                                   bias: weightData.bias,
+                                   id: node.nodeDef.name)
+            }
         }
         mappers[Constants.Ops.Conv] = convMapper
+        mappers[Constants.Ops.DepthwiseConv] = convMapper
```

Comment:
Maybe don't add the `DepthwiseConv` if it's prior to ios11 ? So that it will appear as an unsupported layer?

Reply:
That is a good idea. The other day I was wondering why the result was rubbish until I realized iOS 10 was at fault.